mep/accounts/migrations/0028_temp_precision_event.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.20 on 2019-06-12 14:40
+from __future__ import unicode_literals
+
+from django.db import migrations, transaction
+import mep.accounts.partial_date
+
+
+def copy_precisions(apps, schema_editor):
+    """Copy Borrow and Purchase date precisions over to Event"""
+    Event = apps.get_model('accounts', 'Event')
+
+    for related_name in ('purchase', 'borrow'):
+        # get all events with purchase and borrow events and
+        # pull over related fields to temporary ones on Event
+        events = Event.objects\
```

Comment:
I think you can use a queryset `update` and an [F() expression](https://docs.djangoproject.com/en/2.2/ref/models/expressions/#f-expressions) to do this, which should be a lot faster and more efficient.

Reply:
As you said - an `F` expression actually doesn't work because it doesn't support selecting content across join. I think it will work with a `subquery`.

Reply:
Please also make the data migration reversible, to prevent any unnecessary loss of data if we ever have to backup migrations. Most likely to only come up in dev, but since the reverse logic isn't that hard it would be good to have.

Reply:
Sadly, `subquery` also runs aground. MySQL can't update the table that the SELECT is from using an UPDATE -- and the select for `Borrow`, `Purchase` etc. always ends up being from `Event` (tried flipping the Subquery every way I could think of).

The error code, if you want to take a look and see if I missed anything is `1093` with the text `You can't specify target table`. A join using SQL sidesteps the problem but steps outside the ORM.

Reversible migration written and tested.

Reply:
@bwhicks Thanks for investigating - even if we can't use it here because of the multi-table inheritance it seems like a useful thing to know about, and we may be able to use it in future.

Here's what I tried when I was looking into it before; is it different than your approach? I tested this with your current code and the tests pass. I'm using mariadb for dev, if that makes any difference.

```
borrows = Borrow.objects.filter(event_ptr_id=OuterRef('pk'))
Event.objects.filter(borrow__isnull=False) \
        .update(temp_start_date_precision=Subquery(borrows.values('start_date_precision')),
                temp_end_date_precision=Subquery(borrows.values('end_date_precision')))
```

Reply:
Sadly, this does fail on Maria 5.5, which is what production uses (as does QA). My instinct in that case is to keep the slower option that uses the ORM and should be mostly background agnostic.

Reply:
Ok, thanks for testing to confirm. I'll update my dev environment!